src/compile/cache/mod.rs

```diff
@@ -51,10 +68,58 @@ pub fn cache_dir() -> Result<PathBuf, CacheDirNotSupported> {
 	Ok(env::temp_dir().join("cairo-foundry-tests"))
 }
 
-fn read_cache_file(path: &PathBuf) -> Result<Cache, CacheError> {
-	let file = read_to_string(path)?;
-	let data = serde_json::from_str::<Cache>(file.as_str())?;
-	Ok(data)
+// TODO: support cashing compile errors to avoid recompiling files that failed before
+#[derive(Serialize, Deserialize, Debug)]
+pub struct CompileCacheItem {
+	pub hash: u64,
+	// TODO: make program_json: ProgramJson, we're not using it right now because
+	// it doesn't implement Serialize
+	pub program_json: Value,
+}
+
+impl PartialEq for CompileCacheItem {
+	fn eq(&self, other: &Self) -> bool {
+		self.hash == other.hash
+	}
+}
+
+impl CompileCacheItem {
+	pub fn read(path: &PathBuf) -> Result<CompileCacheItem, CacheError> {
+		let file_content = read_to_string(path)
+			.map_err(|e| CacheError::ReadFile(path.as_path().display().to_string(), e))?;
+
+		let data = serde_json::from_str::<CompileCacheItem>(file_content.as_str())
+			.map_err(|e| CacheError::DeserializeError(file_content, e))?;
+		Ok(data)
+	}
+
+	pub fn write(&self, cache_path: &PathBuf) -> Result<(), CacheError> {
+		let file = File::create(cache_path)
+			.map_err(|e| CacheError::FileCreation(cache_path.as_path().display().to_string(), e))?;
+
+		serde_json::to_writer(file, self).map_err(CacheError::SerializeError)
+	}
+}
+
+pub fn get_compile_cache_path(path_to_cairo_file: &PathBuf) -> Result<PathBuf, CacheError> {
+	// Retrieve only the file name to create a clean compiled file name.
+	let filename = path_to_cairo_file.file_stem().and_then(|f| f.to_str()).ok_or_else(|| {
+		CacheError::StemlessFile(path_to_cairo_file.as_path().display().to_string())
+	})?;
+
+	let path_hash = hash(path_to_cairo_file);
+
+	// Build path to save the  compiled file
+	let mut cache_path = cache_dir()?;
+	cache_path.push(CAIRO_FOUNDRY_CACHE_DIR);
```

Comment:
yup you did replace the old cache dir (CAIRO_FOUNDRY_COMPILED_CONTRACT_DIR) by this one.
It's ok for me, as long as you clean-up the code to get rid of `CAIRO_FOUNDRY_COMPILED_CONTRACT_DIR`